Framework/Algorithms/src/SumSpectra.cpp

```diff
@@ -76,6 +76,40 @@ void SumSpectra::init() {
                   " are removed before the spectra are summed.");
 }
 
+/*
+* Validate the input parameters
+* @returns map with keys corresponding to properties with errors and values
+* containing the error messages.
+*/
+std::map<std::string, std::string> SumSpectra::validateInputs() {
+  // create the map
+  std::map<std::string, std::string> validationOutput;
+
+  // check StartWorkspaceIndex in range
+  MatrixWorkspace_const_sptr localworkspace = getProperty("InputWorkspace");
+  m_numberOfSpectra = static_cast<int>(localworkspace->getNumberHistograms());
+  int maxIndex = getProperty("EndWorkspaceIndex");
+  if (maxIndex != EMPTY_INT() && maxIndex > m_numberOfSpectra) {
+    validationOutput["EndWorkspaceIndex"] =
+        "Selected maximum workspace index is greater than available spectra.";
+  }
+
+  // check ListOfWorkspaceIndices in range
+  const std::vector<int> indices_list = getProperty("ListOfWorkspaceIndices");
+  this->m_indices.clear();
+  // only if specified
+  if (indices_list.size() > 0) {
```

Comment:
Please check `!indices_list.emtpy()` instead of comparison to `0`. It conveys the intention better.